src/pantry.js

```diff
@@ -1,7 +1,97 @@
 class Pantry {
   constructor(userIngredients) {
     this.contents = userIngredients;
+    this.filtered
+    this.goodAmount;
+    this.notEnough = []
+  }
+
+  determineIfUserHasEnoughIngredients(recipe) {
+    let AllIngredientInPantryIds = this.contents.map(ing =>{
+      return ing.ingredient 
+    })
+
+     this.filtered = recipe.ingredients.filter(ingredient => {
+      return AllIngredientInPantryIds.includes(ingredient.id)
+    })
+
+     this.goodAmount =  this.filtered.filter(ing =>{
+      if (AllIngredientInPantryIds.indexOf(ing.id) > -1) {
+        if (!this.contents[AllIngredientInPantryIds.indexOf(ing.id)].amount >= ing.quantity.amount) {
+          this.notEnough.push(ing)
+        }
+        return this.contents[AllIngredientInPantryIds.indexOf(ing.id)].amount >= ing.quantity.amount
+      }
+    })
+    return this.goodAmount.length === this.filtered.length
+  }
+
+  findAmountMissing(recipe) {
+//invoke determineIfUser... ()recipe
+    //if returns false => 
+    //find which ingredients are not satisfied by pantry
+  if (!this.determineIfUserHasEnoughIngredients(recipe)) { 
+    return this.notEnough.length
+    }
   }
 }
 
 export default Pantry;
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+// let pantryIngredientIds = this.contents.map(item => item)
```

Comment:
do we need this pseudocode anymore?  let's make sure we update this file without comments when it's ready!